File: CardGame.Application/DTOs/PlayCardRequestDto.cs
using System.ComponentModel.DataAnnotations;

namespace CardGame.Application.DTOs;

public class PlayCardRequestDto
{
    /// <summary>
    /// The unique ID of the specific card instance being played from the player's hand.
    /// </summary>
    [Required]
    public string CardId { get; set; } = string.Empty;

    /// <summary>
    /// Optional: The ID of the player being targeted by the card effect (if applicable).
    /// </summary>
    public Guid? TargetPlayerId { get; set; }

    /// <summary>
    /// Optional: The type of card being guessed (used only when playing a Guard).
    /// Represented as an integer (rank value).
    /// </summary>
    public int? GuessedCardType { get; set; } // Use string for simplicity in DTO
}

File: CardGame.Web/ClientApp/src/app/features/game/models/playCardRequestDto.ts
/**
 * Represents the data sent to the backend when a player plays a card.
 * Matches the backend PlayCardRequestDto.
 */
export interface PlayCardRequestDto {
  /**
   * The unique ID of the specific card instance being played from the player's hand.
   */
  cardId: string; // Guid as string

  /**
   * Optional: The ID of the player being targeted by the card effect (if applicable).
   */
  targetPlayerId?: string | null; // Guid as string | null

  /**
   * Optional: The type of card being guessed (used only when playing a Guard).
   * Represented as a string (e.g., "Priest", "Baron").
   * Consider sending the integer value if CardType uses values on backend.
   */
  guessedCardType?: number | null;
}
